File: tests/test_merchant.py
import unittest
from usecases.merchant_usecase import MerchantUseCase
from datastore.local_datastore import LocalDataStore


class TestMerchant(unittest.TestCase):
    _merchant_use_case = MerchantUseCase(LocalDataStore())

    def test_1(self):
        """New Merchant"""
        tests = [
            ('m1', 'm1@email.com', '1.5%', True),
            ('m2', 'm2@email.com', '1', False)
        ]

        for test in tests:
            result = self._merchant_use_case.new_merchant(test[0], test[1], test[2])
            self.assertEqual(result['success'], test[3])

    def test_2(self):
        """Update Merchant Discount Percentage"""
        tests = [
            ('m1', '1%', True),
            ('m2', '1', False)
        ]
        for test in tests:
            result = self._merchant_use_case.update_merchant_discount_percentage(test[0], test[1])
            self.assertEqual(result['success'], test[2])

    def test_3(self):
        """Report Discount"""
        tests = [
            ('m1', True),
            ('m2', False)
        ]
        for test in tests:
            result = self._merchant_use_case.report_discount(test[0])
            self.assertEqual(result['success'], test[1])

File: java/src/main/java/com/paylater/tests/TestMerchant.java
package com.paylater.tests;

import com.paylater.repository.LocalDatabase;
import com.paylater.usecases.MerchantUseCase;
import com.paylater.utils.ResponseValue;
import junit.framework.TestCase;

public class TestMerchant extends TestCase {
    private final MerchantUseCase merchantUseCase = new MerchantUseCase(LocalDatabase.getInstance());

    // New Merchant
    public void test1() {
        System.out.println("Running New Merchant");

        ResponseValue result1 = merchantUseCase.newMerchant("m11", "m11@email.com", "1.5%");
        assertTrue(result1.isSuccess());

        ResponseValue result2 = merchantUseCase.newMerchant("m11", "m22@email.com", "1%");
        assertFalse(result1.isSuccess());
    }

    // Update Merchant Discount Percentage
    public void test2() {
        System.out.println("Running Update Merchant Discount Percentage");

        ResponseValue result1 = merchantUseCase.updateMerchantDiscountPercentage("m11", "1");
        assertTrue(result1.isSuccess());

        ResponseValue result2 = merchantUseCase.updateMerchantDiscountPercentage("m222", "1.5%");
        assertFalse(result1.isSuccess());
    }

    // Report Discount
    public void test3() {
        System.out.println("Running Report Discount");

        ResponseValue result1 = merchantUseCase.reportDiscount("m11");
        assertTrue(result1.isSuccess());

        ResponseValue result2 = merchantUseCase.reportDiscount("m22");
        assertFalse(result1.isSuccess());
    }
}
